cylc/flow/scheduler.py

```diff
@@ -716,8 +716,9 @@ async def start(self):
             self.server.thread.start()
             barrier.wait()
 
-            self._configure_contact()
             await self.configure()
+            self.task_events_mgr.uuid_str = self.uuid_str
+            self._configure_contact()
```

Comment:
`_configure_contact` should happen before `configure`.

https://github.com/cylc/cylc-flow/commit/0aadc909aae6a687545e665854340ef1cb794a54

Reply:
It's nasty to do it this way around because another scheduler can startup for the same workflow whilst Cylc is loading the configuration which lead to a reported bug where three schedulers were running for the same workflow simultaneously.

Reply:
* Various template variables are required at various stages of startup.
* `Scheduler` logic is likely sensitive to the timing of `load_workflow_params_and_tmpl_vars` during startup.

> Can we accept the extra database call that I proposed in an earlier iteration to simply get the UUID at this point?

Does this mean we need two database calls to fetch these variables, no.

E.G:

```
def load_db_params(self):
    params = []
    self.workflow_db_mgr.select_workflow_params(params.append)
    return params

def start(self):
    ...
    params = self.load_db_params()
    ...
    self._configure_contact(..., params)
    self.configure(..., params)

def load_workflow_params_and_tmpl_vars(self, params):
    for key, value in params:
        ...
```